tests/onnx/test_onnx_model_export.py

```diff
@@ -0,0 +1,198 @@
+import sklearn.datasets as datasets
+import pandas as pd
+import numpy as np
+import yaml
+
+import mlflow
+import mlflow.keras
+import mlflow.pyfunc.scoring_server as pyfunc_scoring_server
+from mlflow import pyfunc
+from mlflow.tracking.artifact_utils import _get_model_log_dir
+from mlflow.utils.environment import _mlflow_conda_env
+from mlflow.utils.model_utils import _get_flavor_configuration
+from tests.projects.utils import tracking_uri_mock  # pylint: disable=unused-import
+
+import onnx
+import onnxmltools
+import mlflow.onnx
+
+pytestmark = pytest.mark.skipif(
+    (sys.version_info < (3, 6)),
+    reason="Tests require Python 3 to run!")
+
+
+@pytest.fixture(scope='module')
+def data():
+    iris = datasets.load_iris()
+    data = pd.DataFrame(data=np.c_[iris['data'], iris['target']],
+                        columns=iris['feature_names'] + ['target'])
+    y = data['target']
+    x = data.drop('target', axis=1)
+    return x, y
+
+
+@pytest.fixture(scope='module')
+def model(data):
+    x, y = data
+    model = Sequential()
+    model.add(Dense(3, input_dim=4))
+    model.add(Dense(1))
+    model.compile(loss='mean_squared_error', optimizer='SGD')
+    model.fit(x, y)
+    return model
+
+
+@pytest.fixture(scope='module')
+def onnx_model(model):
+    return onnxmltools.convert_keras(model)
+
+
+@pytest.fixture(scope='module')
+def predicted(model, data):
+    return model.predict(data[0])
+
+
+@pytest.fixture
+def model_path(tmpdir):
+    return os.path.join(tmpdir.strpath, "model")
+
+
+@pytest.fixture
+def onnx_custom_env(tmpdir):
+    conda_env = os.path.join(str(tmpdir), "conda_env.yml")
+    _mlflow_conda_env(
+        conda_env,
+        additional_conda_deps=["pytest", "keras"],
+        additional_pip_deps=["onnx", "onnxmltools", "onnxruntime"])
+    return conda_env
+
+
+@pytest.mark.large
+def test_model_save_load(onnx_model, model_path, data, predicted):
+    x, y = data
+    mlflow.onnx.save_model(onnx_model, model_path)
+
+    # Loading ONNX model
+    model_loaded = mlflow.onnx.load_model(model_path)
+    onnx.checker.check_model(model_loaded)
+
+    # Loading pyfunc model
+    pyfunc_loaded = mlflow.pyfunc.load_pyfunc(model_path)
+    assert np.allclose(pyfunc_loaded.predict(x).values, predicted, rtol=1e-05, atol=1e-05)
+
+
+@pytest.mark.large
+def test_model_log(tracking_uri_mock, onnx_model, data,
+                   predicted):  # pylint: disable=unused-argument
+    x, y = data
+    # should_start_run tests whether or not calling log_model() automatically starts a run.
+    for should_start_run in [False, True]:
+        try:
+            if should_start_run:
+                mlflow.start_run()
+            mlflow.onnx.log_model(onnx_model=onnx_model, artifact_path="onnx_model")
+
+            # Load model
+            model_loaded = mlflow.onnx.load_model(
+                "onnx_model",
+                run_id=mlflow.active_run().info.run_uuid)
+            onnx.checker.check_model(model_loaded)
```

Comment:
The `mlflow.onnx.load_model` method already performs the `check_model` test. We should verify that this is the case by mocking the `check_model` method and asserting that the mock is invoked when calling `mlflow.onnx.load_model`. If another test already verifies that `mlflow.onnx.load_model` calls `check_model`, then we can remove this line.

Reply:
Changed tests to use mock.